instrumentation/mongodb/src/main/java/brave/mongodb/TraceMongoCommandListener.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.mongodb;
+
+import brave.Span;
+import brave.internal.Nullable;
+import brave.propagation.ThreadLocalSpan;
+import com.mongodb.MongoException;
+import com.mongodb.MongoSocketException;
+import com.mongodb.connection.ConnectionDescription;
+import com.mongodb.connection.ConnectionId;
+import com.mongodb.event.CommandFailedEvent;
+import com.mongodb.event.CommandListener;
+import com.mongodb.event.CommandStartedEvent;
+import com.mongodb.event.CommandSucceededEvent;
+import org.bson.BsonDocument;
+import org.bson.BsonValue;
+import org.bson.json.JsonWriterSettings;
+
+import java.net.InetSocketAddress;
+import java.util.Set;
+
+/**
+ * A MongoDB command listener that will report via Brave how long each command takes and other information about the
+ * commands.
+ *
+ * Implementation notes regarding the <b>synchronous</b> MongoDB clients ({@link com.mongodb.MongoClient} and
```

Comment:
move this stuff to RATIONALE.md as it is easier to read. you can make a comment to look at that if you like